File: code_challenge_solution/javascript/11 Binary search.js
```javascript
function binarySearch(arr, target) {
  let left = 0;
  let right = arr.length - 1;

  while (left <= right) {
    let mid = Math.floor((left + right) / 2);

    if (arr[mid] === target) {
      return mid;
    } else if (arr[mid] < target) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }

  return -1;
}

// Example usage:
const array = [2, 5, 8, 12, 16, 23, 38, 56, 72, 91];
const target = 23;
const result = binarySearch(array, target);
console.log(result); // Output: 5
```

File: code_challenge_solution/c#/11 Binary search.cs
Sure! Here is the code for a binary search algorithm in C#:

```csharp
using System;

class BinarySearchAlgorithm
{
    static int BinarySearch(int[] array, int target)
    {
        int left = 0;
        int right = array.Length - 1;

        while (left <= right)
        {
            int mid = (left + right) / 2;

            if (array[mid] == target)
            {
                return mid;
            }
            else if (array[mid] < target)
            {
                left = mid + 1;
            }
            else
            {
                right = mid - 1;
            }
        }

        return -1;
    }

    static void Main(string[] args)
    {
        int[] array = { 2, 5, 8, 12, 16, 23, 38, 56, 72, 91 };
        int target = 23;

        int result = BinarySearch(array, target);

        if (result != -1)
        {
            Console.WriteLine("Element found at index " + result);
        }
        else
        {
            Console.WriteLine("Element not found in array");
        }
    }
}
```

In this code, the `BinarySearch` method takes an array and a target value as inputs and returns the index of the target value in the array using the binary search algorithm. The `Main` method initializes an array and a target value, calls the `BinarySearch` method, and outputs the result.